src/errors.rs

```diff
@@ -0,0 +1,95 @@
+use tonic::{transport::Error as TonicError, Status};
+
+#[derive(Debug, thiserror::Error)]
+pub enum EigenClientError {
+    #[error(transparent)]
+    EthClientError(#[from] EthClientError),
+    #[error(transparent)]
+    VerificationError(#[from] VerificationError),
+    #[error("Private Key Error")]
+    PrivateKeyError,
+    #[error(transparent)]
+    SecpError(#[from] secp256k1::Error),
+    #[error(transparent)]
+    HexError(#[from] hex::FromHexError),
+    #[error(transparent)]
+    RlpError(#[from] rlp::DecoderError),
+    #[error(transparent)]
+    TonicError(#[from] TonicError),
```

Comment:
Think we can remove the `Error` suffix everywhere? I personally feel like its redundant. `EigenClientError::PrivateKey` reads better than `EigenClientError::PrivateKeyError` I feel.